serialize/serialize.go

```diff
@@ -65,12 +66,21 @@ func ToBytes(tracer opentracing.Tracer, spanContext opentracing.SpanContext, req
 
 	var writer bytes.Buffer
 	err = protocol.Binary.Encode(wireValue, &writer)
-	return writer.Bytes(), err
+
+	// prepend bytes with 0 to indicate thrift serialization
+	yarpcBytes := append([]byte{0}, writer.Bytes()...)
```

Comment:
Your comment should somehow indicate that we are versioning the payload with an integer in the first position.